### PR TITLE
add shared expense feature now working.

### PR DESCRIPTION
**Pull request by nikoootine**
customized auto complete in the add expense dialog

### CODE REVIEW

File: app/Http/Controllers/DashboardController.php [hunk 1]
```diff
@@ -20,30 +23,41 @@ public function __construct()
...
-                $newSharedExpense->amount_owed=$req->expOwedAmount;
-                $newSharedExpense->comments=$req->expOwerComments;
-                $newSharedExpense->secondary_username=$req->expOwerUsername;
-                $newSharedExpense->save();
+            if(isset($req->username) || isset($req->expOwedAmount)) {
+
+              // TODO Set comment to nullable in database
+              $this->validate($req, [
+                  'expOwedAmount'   => 'required|numeric|min:1',
+                  'username'        => 'exists:users|required|max:255',
```
Comment (nikoootine) [inline app/Http/Controllers/DashboardController.php]: i put username to match the column in the users table. other column names throws an error

File: database/migrations/2017_04_04_202103_UpdateSharedExpensesDateSettled.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
+<?php
```
Comment (nikoootine) [inline database/migrations/2017_04_04_202103_UpdateSharedExpensesDateSettled.php]: This file is not needed anymore as Vince also made changes to the tables in his previous branch

File: resources/views/layouts/app.blade.php [hunk 3]
```diff
@@ -33,7 +36,7 @@
         });
     </script>
 </head>
-<body>
+<body onload="init()">
```
Comment (nikoootine) [inline resources/views/layouts/app.blade.php]: Note to @SplitSet3R/front-end  I suggest everyone to put anything that requires onload in init() function in custom.js

**Comment by nikoootine** [comment]
Yeah. This uses an outdated database. The Feature Notifications branch made changes to it thats why. I'll update this branch

**Comment by itsmattli** [comment]
was my fault, missed a migration on the expenses table.  I added a preseeded date of current date to the modal.  Should be good now.

@SplitSet3R/front-end just as a note, you guys have two copies of the add expense modal, one in the dashboard and one in the sidebar. (not sure if you guys already know)

**Pull request merged**